### PR TITLE
add auth , intial setup and create sidebar

### PR DESCRIPTION
**Pull request by archit7448**
### What I did : 

- Add auth 
- create sidebar  
- [files to check](https://github.com/archit7448/SocialGram/pull/1/files?file-filters%5B%5D=.css&file-filters%5B%5D=.js&file-filters%5B%5D=.jsx&file-filters%5B%5D=.png&show-deleted-files=true&show-viewed-files=true)

### CODE REVIEW

File: src/Utility/CSS/main.css [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+:root {
+  --color-primary-first: black;
+  --color-primary-second: #0096FF;
+  --color-background-primary: white;
+  --color-icon: rgb(44, 53, 57);
```
Comment (amartya-rs) [inline src/Utility/CSS/main.css:11]: suggestion: for consistency try to use only one color code, either hex or rgb

File: src/index.js [hunk 2]
```diff
@@ -2,14 +2,21 @@ import React from "react";
...
 import "./index.css";
 import App from "./App";
+import { store } from "./store";
 import { makeServer } from "./server";
-
+import { BrowserRouter } from "react-router-dom";
+import { Provider } from "react-redux";
 // Call make Server
 makeServer();
+console.log(store.getState())
```
Comment (amartya-rs) [inline src/index.js:11]: remove console logs. You can try the redux devtools extension for keeping a track of the states.

File: src/pages/Explore/Explore.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { useEffect } from "react";
+import { Sidebar } from "../../Components";
+import axios from "axios";
+import { useDispatch, useSelector } from "react-redux";
+import { AddPost } from "../../reducer/postSlice";
+
+export const Explore = () => {
+  const { post } = useSelector((store) => store.posts);
+  console.log(post);
+  const dispatch = useDispatch();
```
Comment (amartya-rs) [inline src/pages/Explore/Explore.jsx:10]: remove console logs.

File: src/pages/Explore/Explore.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import { useDispatch, useSelector } from "react-redux";
+import { AddPost } from "../../reducer/postSlice";
+
+export const Explore = () => {
+  const { post } = useSelector((store) => store.posts);
+  console.log(post);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    (async () => {
+      try {
```
Comment (amartya-rs) [inline src/pages/Explore/Explore.jsx:13]: for api call use createAsyncThunk provided by redux

File: src/reducer/postSlice.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  posts: [],
+};
+
+const postSlice = createSlice({
+  name: "posts",
+  initialState,
+  reducers: {
+    AddPost: (state, action) => {
+      console.log(state.posts);
+      console.log(action.payload);
```
Comment (amartya-rs) [inline src/reducer/postSlice.js]: remove console logs.

File: src/App.js [hunk 6]
```diff
@@ -1,7 +1,28 @@
...
+} from "./pages/index";
 
 function App() {
-  return <div className="App"></div>;
+  return (
+    <div className="App">
+      <Routes>
+        <Route path="/" element={<SignIn />} />
+        <Route path="/home" element={<Explore />} />
+        <Route path="/SignUp" element={<SignUp />} />
```
Comment (amartya-rs) [inline src/App.js:19]: path should be in small case. `/signup`

**Review by amartya-rs — APPROVED** [review]
In overall code looks great to me. Just look into the suggestions.

File: src/pages/Explore/Explore.jsx (hunk 4 above)
Comment (archit7448) [reply src/pages/Explore/Explore.jsx:13]: yeah, but I am not using any loader, so It doesn't make any sense

**Review by itsmebipul00 — APPROVED** [review]
Good to merge👍

**Pull request merged**